agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -337,29 +341,65 @@ func (dg *dockerGoClient) time() ttime.Time {
 // Pulls image manifest from the registry
 func (dg *dockerGoClient) PullImageManifest(
 	ctx context.Context, imageRef string, authData *apicontainer.RegistryAuthenticationData,
-) (registry.DistributionInspect, error) {
+) (registry.DistributionInspect, apierrors.NamedError) {
 	// Get auth creds
 	sdkAuthConfig, err := dg.getAuthdata(imageRef, authData)
 	if err != nil {
-		return registry.DistributionInspect{}, wrapPullErrorAsNamedError(imageRef, err)
+		return registry.DistributionInspect{}, wrapManifestPullErrorAsNamedError(imageRef, err)
 	}
 	encodedAuth, err := registry.EncodeAuthConfig(sdkAuthConfig)
 	if err != nil {
-		return registry.DistributionInspect{}, wrapPullErrorAsNamedError(imageRef, err)
+		return registry.DistributionInspect{}, wrapManifestPullErrorAsNamedError(imageRef, err)
 	}
 
-	// Get an SDK Docker Client and call Distribution API
```

Comment:
Q: Isn't this comment still true regarding calling the Distribution API?

Reply:
I initially had a similar thought, but looks like this information was moved to L361.

Reply:
Oh I see. Disregard then!